aptos-move/block-executor/src/executor.rs

```diff
@@ -284,7 +292,7 @@ where
             versioned_cache.delayed_fields().remove(&id, idx_to_execute);
         }
 
-        if !last_input_output.record(idx_to_execute, read_set, result) {
+        if !last_input_output.record(idx_to_execute, read_set, result, resource_write_set) {
```

Comment:
Why do we need to start recording `resource_write_set` in last_input_output now?

Reply:
We were cloning the whole write set twice, now we cache it here.